modules/dispatcher/dispatcher.c

```diff
@@ -1360,15 +1361,30 @@ mi_response_t *ds_mi_reload_1(const mi_params_t *params,
 {
 	ds_partition_t *partition;
 	str partname;
+	str inherit_state;
+	int is_inherit_state = INHERIT_STATE_YES;
 
 	if (get_mi_string_param(params, "partition", &partname.s, &partname.len) < 0)
 		return init_mi_param_error();
 
+	if (get_mi_string_param(params, "inherit_state", &inherit_state.s, &inherit_state.len) >= 0) {
+		LM_DBG("inherit_state is: %s \n", inherit_state.s);
+
+		if (inherit_state.s[0] == '0' || inherit_state.s[0] == 'n' || inherit_state.s[0] == 'N') {
+			is_inherit_state = INHERIT_STATE_NO;
+		}
+		else if (inherit_state.s[0] == '1' || inherit_state.s[0] == 'y' || inherit_state.s[0] == 'Y') {
+			is_inherit_state = INHERIT_STATE_YES;
+		}
```

Comment:
Maybe you can add here a warning log saying that the `inherit_state` values was not recognized, so ignored.